.github/workflows/build.yml

```diff
@@ -0,0 +1,98 @@
+name: Build
+
+on: [push]
+
+jobs:
+  Linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential cmake libgl-dev libxrandr-dev libxinerama-dev libxcursor-dev libxi-dev
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Build
+        run: |
+          cmake -B bin -DCMAKE_BUILD_TYPE=Release .
+          cmake --build bin
+      - name: Publish artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: OpenNFS-Linux
+          path: |
+            bin/OpenNFS
+            resources
+            shaders
+            LICENSE
+            README.md
+
+  Windows:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Build
+        run: |
+          cmake -B bin -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=C:\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -A x64 .
+          cmake --build bin --config Release
+          cp bin\Release\OpenNFS.exe bin\OpenNFS.exe
+          cp LICENSE LICENSE.txt
+      - name: Publish artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: OpenNFS-Windows
+          path: |
+            bin/OpenNFS.exe
+            resources
+            shaders
+            LICENSE.txt
+            README.md
+
+  MacOS:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Build
+        run: |
+          cmake -B bin -DCMAKE_BUILD_TYPE=Release .
```

Comment:
`ninja` as well

Reply:
I added `--parallel 3` now, which is faster than using Ninja.

Here a comparison:
- Parallel option: https://github.com/sharkwouter/OpenNFS/actions/runs/11840113594
- Ninja: https://github.com/sharkwouter/OpenNFS/actions/runs/11840176830